### DISCUSSION TITLE
SQLite Version and sqlite3.dll (binary) location

### DISCUSSION THREAD

**Discussion by FundComps** [pawelsalawa/sqlitestudio#4004] [Q&A]
First, congratulations on getting this released.  I'm new to it, but so far, I'm impressed!

In a previous discussion thread from December, you'd indicated that you could downgrade to a slightly earlier version of the core sqlite "engine" if required, by replacing the sqlite3.dll (I'm on Windows 10) that should be located in the SQLiteStudio root directory.

However, I'm not seeing that DLL in your installation, so I'm a bit lost as where the core SQLLite "engine" is being installed?  I just would like to better understand the linkage between the studio and the core sqlite engine, and knowing that would be very helpful!

Thanks...

**Comment by pawelsalawa** [comment]
Hi,

Unfortunately this is no longer an option with 3.3.0 version. SQLite is now built-in, compiled together with entire application as one. There were few reasons for that, among which it's causing less maintenance effort across different operating systems.

Why would you need to downgrade the engine?

**Comment by FundComps** [comment]
Hi,

OK, that makes sense.

The use case for me is hypothetical:  I do my development and all my testing with this version of SQLITE (version 3.34.1).  I do quite a bit of development and testing with this version.  I'm in production....  Then 6 months later SQLITE releases a patch, and you upgrade your software as well that bundles their new version.   I try to upgrade, and find a breaking change in the new version of SQLITE with my application code, so I want to stick with 3.34.1, but at the same time I'd love to take a new version of your studio. 

I realize this may be an edge case given that SQLITE appears to be quite stable.  And I also get that trying to maintain your software with too many versions of SQLite could be a real headache!  For now, I won't worry to much about it, as I'm hopefully just over-thinking this.

And, thanks for the extremely prompt reply!

Lee
